src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyGenerator.java

```diff
@@ -26,8 +26,8 @@
      */
     public static final String APPENDIX_CHARACTERS = "abcdefghijklmnopqrstuvwxyz";
     private static final Logger LOGGER = LoggerFactory.getLogger(BibtexKeyGenerator.class);
-    private static final String KEY_ILLEGAL_CHARACTERS = "{}(),\\\"#~^':`";
-    private static final String KEY_UNWANTED_CHARACTERS = "{}(),\\\"";
+    private static final String KEY_ILLEGAL_CHARACTERS = "{}(),\\\"-#~^':`";
```

Comment:
@chelseyong Do you, by chance, remember why you introduced `-` here? We have issues with that. See https://github.com/JabRef/jabref/issues/6295 for details.

Reply:
I think it is due to a suggestion made in the [issue](https://github.com/JabRef/jabref/issues/4709).
> Al-Ketan, 2019 -> Al-19 (dash in bibtex key is awkward, should probably be ignored in favor of the next letter and become AlK19)

Reply:
Thank you for your reply. I Think, there was some misunderstanding at https://github.com/JabRef/jabref/issues/4709#issuecomment-468700259

> Well, dash is a **perfectly legal char**

We will fix it at https://github.com/JabRef/jabref/pull/6300